lib/sha256.c

```diff
@@ -224,6 +224,7 @@ static void SHA256_Update(SHA256_CTX *ctx,
 static void SHA256_Final(unsigned char *digest, SHA256_CTX *ctx)
 {
   unsigned long length;
+  length = 0;
```

Comment:
Seem fair, but while fixing this code shouldn't the code also check the return code from CryptGetHashParam() ?